05_concurrency/homework/main_test.go

```diff
@@ -0,0 +1,230 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+)
+
+type Cache interface {
+	Set(key string, value interface{}, ttl time.Duration)
+	Get(key string) (interface{}, bool)
+	Delete(key string)
+}
+
+type Value struct {
+	value interface{}
+	ttl   time.Duration
+	exp   time.Time
+}
+
+func (v Value) IsExpired() bool {
+	return v.exp.Sub(time.Now()) < 0
```

Comment:
Go's std lib has useful methods to compare time values (e.g. https://pkg.go.dev/time#Time.Before)